common/src/main/java/com/wynntils/features/user/tooltips/ItemCompareFeature.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.features.user.tooltips;
+
+import com.mojang.blaze3d.vertex.PoseStack;
+import com.wynntils.core.features.UserFeature;
+import com.wynntils.core.features.properties.FeatureInfo;
+import com.wynntils.core.features.properties.RegisterKeyBind;
+import com.wynntils.core.keybinds.KeyHolder;
+import com.wynntils.mc.event.InventoryKeyPressEvent;
+import com.wynntils.mc.event.ItemTooltipRenderEvent;
+import com.wynntils.mc.event.ScreenClosedEvent;
+import com.wynntils.mc.event.SlotRenderEvent;
+import com.wynntils.mc.render.RenderUtils;
+import com.wynntils.mc.utils.McUtils;
+import com.wynntils.utils.objects.CommonColors;
+import com.wynntils.wc.custom.item.GearItemStack;
+import java.util.List;
+import java.util.Optional;
+import net.minecraft.client.gui.screens.inventory.AbstractContainerScreen;
+import net.minecraft.world.inventory.Slot;
+import net.minecraft.world.item.ItemStack;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import org.lwjgl.glfw.GLFW;
+
+@FeatureInfo(stability = FeatureInfo.Stability.STABLE, category = "Item Tooltips")
+public class ItemCompareFeature extends UserFeature {
+    @RegisterKeyBind
+    private final KeyHolder toggleCompareModeKeybind =
+            new KeyHolder("Compare mode", GLFW.GLFW_KEY_SPACE, "Wynntils", true, () -> {});
+
+    @RegisterKeyBind
+    private final KeyHolder compareSelectKeybind =
+            new KeyHolder("Select for comparing", GLFW.GLFW_KEY_C, "Wynntils", true, () -> {});
+
+    private static GearItemStack comparedItem = null;
+    private static boolean compareToEquipped = false;
+
+    @SubscribeEvent
+    public void onRenderSlot(SlotRenderEvent.Pre event) {
+        Slot slot = event.getSlot();
+        if (slot.getItem() == comparedItem) {
+            RenderUtils.drawArc(CommonColors.LIGHT_BLUE, slot.x, slot.y, 200, 1, 6, 8);
+        }
+    }
+
+    @SubscribeEvent
+    public void onItemTooltipRenderEvent(ItemTooltipRenderEvent.Post event) {
+        if (!(McUtils.mc().screen instanceof AbstractContainerScreen<?> abstractContainerScreen)) return;
```

Comment:
Thinking a bit more of this, I think that `null instanceof AnyClass` is always true, so maybe we should combine this with a null check as well. Otoh, you can't really get a tooltip event unless in some kind of screen, can you?

Reply:
Isn't the tooltip drawn with chat items?

Reply:
That uses a renderutils method iirc.

Reply:
chat items use the vanilla tooltip rendering method, so it would trigger the event. but at the same time, the screen wouldn't be null, it would be the chat gui